mlir/include/mlir/Dialect/LLVMIR/LLVMIntrinsicOps.td

```diff
@@ -455,7 +456,7 @@ def LLVM_DbgDeclareOp : LLVM_DbgIntrOp<"dbg.declare", "addr",
   let arguments = (ins
     LLVM_AnyPointer:$addr,
     LLVM_DILocalVariableAttr:$varInfo,
-    DefaultValuedAttr<LLVM_DIExpressionAttr, "std::nullopt">:$locationExpr
+    OptionalAttr<LLVM_DIExpressionAttr>:$locationExpr
```

Comment:
This may be a personal preference, but would a DefaultValuedAttr be simpler here for the semantics of the op? That way there's only one way to represent an empty expression, so the user doesn't have to check/maintain when getting/setting.

Reply:
I tried to keep it the way you had it originally, but it just didn't generate valid code. It would generate the assignment without the specified default value (locationExpr = ).

Reply:
```
void GlobalOp::populateDefaultProperties(::mlir::OperationName opName, Properties &properties) {
  ::mlir::Builder odsBuilder(opName.getContext());
  // template is: $_builder.getIntegerAttr($_builder.getIntegerType(32), $0)
  // default is: 0
  if (!properties.addr_space)
    properties.addr_space = odsBuilder.getIntegerAttr(odsBuilder.getIntegerType(32), 0);
  // template is: 
  // default is: ::mlir::LLVM::DIGlobalVariableExpressionAttr()
  if (!properties.dbg_expr)
    properties.dbg_expr = ;
  // template is: ::mlir::LLVM::VisibilityAttr::get($_builder.getContext(), $0)
  // default is: mlir::LLVM::Visibility::Default
  if (!properties.visibility_)
    properties.visibility_ = ::mlir::LLVM::VisibilityAttr::get(odsBuilder.getContext(), mlir::LLVM::Visibility::Default);
}
```

Looks like something is overriding the `constBuilderCall` for the attribute

Reply:
Figured it out and now it has a working default.